pkg/config/runtime_test.go

```diff
@@ -37,14 +42,14 @@ spec:
   nodeConfig:
     metadata:
       name: k0s
-  pid: -1
 `)
 	require.NoError(t, os.WriteFile(rtConfigPath, content, 0644))
 
 	// try to load runtime config and check if it returns an error
 	spec, err := LoadRuntimeConfig(rtConfigPath)
 	assert.Nil(t, spec)
 	assert.ErrorIs(t, err, ErrK0sNotRunning)
+	t.Cleanup(func() { require.NoError(t, spec.Cleanup()) })
```

Comment:
Wouldn't a call to `spec.Cleanup` only be necessary after a successful call to `NewRuntimeConfig`?

Moreover, since `LoadRuntimeConfig` returned an error, there shouldn't be the need to call `Cleanup`, especially as spec is `nil`. So if the goal is to test that `Cleanup` is a no-op when called on a `nil` receiver, I'd rather do this in a separate test case.

Reply:
good points. I'll fix. Thanks!